src/java/org/apache/cassandra/db/guardrails/Guardrail.java

```diff
@@ -89,6 +101,7 @@ protected void fail(String message, @Nullable ClientState state)
         ClientWarn.instance.warn(message);
         // Similarly, tracing will also ignore the message if we're not running tracing on the current thread.
         Tracing.trace(message);
+        GuardrailsDiagnostics.failed(name, decorateMessage(redactedMessage));
 
         if (state != null)
```

Comment:
@adelapena maybe unrelated to this ticket but why are we throwing right here as the last statement in the method? it is little bit counter-intuitive to me.

Reply:
Because we want to emit client warnings, log messages and events before actually aborting the operation.

Reply:
ah ... right right, sure :)

Reply:
Note that the exception is an {{InvalidRequestException}} that won't be printed in logs nor produce particular observable output.